index.html

```diff
@@ -1,100 +1,122 @@
-  var file = this.files[0];
-  if (file) {
-  loadedFile = file;
-  alert('Plik Excel został załadowany i zapisany.');
-      }
-    });
-    
-//tutaj znajduje sie kod do wczytania danych z excela 
-    document.getElementById('select').addEventListener('change', function() {
-      var selectedOption = parseInt(this.value);
-      if (loadedFile) {
-        var reader = new FileReader();
-        reader.onload = function(e) {
-          var data = new Uint8Array(e.target.result);
-          var workbook = XLSX.read(data, { type: 'array' });
-          var worksheet = workbook.Sheets[workbook.SheetNames[0]];
-          var selectedCell = worksheet[XLSX.utils.encode_cell({ r: selectedOption, c: 5 })];
-          var cellValue = selectedCell ? selectedCell.v : 'Brak danych';
-          document.getElementById('wybranyokreg').textContent = "Wybrany okreg wyborczy zawiera " + cellValue + " wyborcow";
-        };
-
-        reader.readAsArrayBuffer(loadedFile);
-      } else {
-        alert('Proszę najpierw załadować plik Excel.');
-      }
-    });
-  </script>
+    <h1>liczba wyborcow w danych okregach wyborczych</h1>
+    <button id="przyciskcsv">Załaduj plik CSV</button> <br>
+    <input type="file" id="filei" accept=".csv" style="display: none;">
+    <select id="select"></select>
+    <div id="wybranyokreg"></div>
+    <div id="kwadratodleglosci"></div>
+
+    <script>
+        var loadedFile;
+        var addedDistricts = [];
+
+        document.getElementById('przyciskcsv').addEventListener('click', function () {
+            document.getElementById('filei').click();
+        });
+
+        document.getElementById('filei').addEventListener('change', function () {
+            var file = this.files[0];
+            if (file) {
+                loadedFile = file;
+                alert('Plik CSV został załadowany i zapisany.');
+                populateDropdown();
+            }
+        });
+
+        document.getElementById('select').addEventListener('change', function () {
+            var selectedOption = this.value;
+            if (loadedFile) {
+                var reader = new FileReader();
+                reader.onload = function (e) {
+                    var csvData = e.target.result;
+                    var rows = csvData.split('\n');
+                    var selectedDistrict = rows[selectedOption].split(';')[0];
+
+                    var districtRows = rows.filter(function (row) {
+                    return row.startsWith(selectedDistrict + ';');
+                    });
 
+                    displayResults(districtRows);
+                    kwadratodleglosci();
+                };
 
+                reader.readAsText(loadedFile);
+            } else {
+                alert('Proszę najpierw załadować plik CSV.');
+            }
+        });
 
+        function kwadratodleglosci() {
+            var selectedOption = document.getElementById('select').value;

```

Comment:
Minor:

I believe you could pass the `selectedOption` as a parameter from line 47. This would avoid the repetition here (you would be reading the `selectedOption` only once and then reusing it throughout the code)

```suggestion
        function kwadratodleglosci(selectedOption) {

```